spec/acceptance/tool_group_controller_spec.rb

```diff
@@ -0,0 +1,312 @@
+# frozen_string_literal: true
+
+require "acceptance_helper"
+
+resource "ToolGroups" do
+  header "Accept", "application/vnd.api+json"
+  header "Content-Type", "application/vnd.api+json"
+
+  let(:raw_post) { params.to_json }
+  let(:authorization) { AuthToken.generic_token }
+  let(:languages) { ["es", "en"] }
+  let(:countries) { ["AR", "ES"] }
+  let(:openness) { [1, 2, 3] }
+  let(:confidence) { [1, 2] }
+  let(:metatool_resource_type) { ResourceType.find_by(name: "metatool") }
+  let(:resource) { FactoryBot.create(:resource, system_id: 1, resource_type: metatool_resource_type, name: "Resource Test One", abbreviation: "test1") }
+  let(:tool_group_first) { ToolGroup.first }
+  let(:tool_group_last) { ToolGroup.last }
+
+  before(:each) do
+    %i[one two three].each do |name|
+      FactoryBot.create(:tool_group, name: name)
+    end
+    FactoryBot.create(:rule_language, tool_group: ToolGroup.first, languages: languages)
+    FactoryBot.create(:rule_country, tool_group: ToolGroup.first, countries: countries)
+    FactoryBot.create(:rule_praxis, tool_group: ToolGroup.first, openness: openness, confidence: confidence)
+    resource
+  end
+
+  after(:each) do
+    RuleCountry.delete_all
+    RuleLanguage.delete_all
+    RulePraxis.delete_all
+    ResourceToolGroup.delete_all
+    ToolGroup.delete_all
+  end
+
+  post "tool-groups" do
+    let(:attrs) do
+      {
+        name: "test",
+        suggestions_weight: "1.0"
+      }
+    end
+
+    let(:attrs_invalid) do
+      {
+        name: "test",
+        suggestions_weight: ""
+      }
+    end
+
+    requires_authorization
+
+    it "create tool group" do
+      do_request data: {type: "tool-group", attributes: attrs}
+      expect(status).to eq(201)
+      expect(JSON.parse(response_body)["data"]).not_to be_nil
+    end
+
+    it "returns error message when tool group is not created" do
+      do_request data: {type: "tool-group", attributes: attrs_invalid}
+
+      expect(status).to eq(400)
+      expect(JSON.parse(response_body)["errors"]).not_to be_empty
+      expect(JSON.parse(response_body)["errors"][0]["detail"]).to eql "Validation failed: Suggestions weight can't be blank"
+    end
+  end
+
+  post "tool-groups/:tool_group_id/tools" do
+    let(:attrs) do
+      {
+        "relationships": {
```

Comment:
`relationships` shouldn't be a child of `attributes`, it should be at the same level as attributes.

```
{
  "attributes": {},
  "relationships": {}
}
```

Reply:
solved